apisix/plugins/error-log-logger.lua

```diff
@@ -0,0 +1,220 @@
+local string = string
+local tostring = tostring
+local buffers
+local loaded_flag
+
+local metadata_schema = {
+    type = "object",
+    properties = {
+        host = {type = "string", default = "127.0.0.1"},
+        port = {type = "integer", minimum = 0, default = 9200},
+        tls = {type = "boolean", default = false},
+        tls_options = {type = "string"},
+        timeout = {type = "integer", minimum = 1, default = 3},
+        keepalive = {type = "integer", minimum = 1, default = 30},
+        name = {type = "string", default = plugin_name},
+        level = {type = "string", default = "WARN"},
+        batch_max_size = {type = "integer", minimum = 0, default = 1000},
+        max_retry_count = {type = "integer", minimum = 0, default = 0},
+        retry_delay = {type = "integer", minimum = 0, default = 1},
+        buffer_duration = {type = "integer", minimum = 1, default = 60},
+        inactive_timeout = {type = "integer", minimum = 1, default = 3},
+    },
+    additionalProperties = false,
+}
+
+
+local log_level = {
+    STDERR =    ngx.STDERR,
+    EMERG  =    ngx.EMERG,
+    ALERT  =    ngx.ALERT,
+    CRIT   =    ngx.CRIT,
+    ERR    =    ngx.ERR,
+    ERROR  =    ngx.ERR,
+    WARN   =    ngx.WARN,
+    NOTICE =     ngx.NOTICE,
+    INFO   =    ngx.INFO,
+    DEBUG  =    ngx.DEBUG
+}
+
+
+local config = {
+}
+
+
+local _M = {
+    version = 0.1,
+    priority = 1091,
+    name = plugin_name,
+    metadata_schema = metadata_schema,
+    timer = nil
+}
+
+
+local function check_metadata_schema(conf)
+    return core.schema.check(metadata_schema, conf)
+end
+
+
+local function send_to_server(data)
+    local res = false
+    local err_msg
+    local sock, soc_err = tcp()
+
+    if not sock then
+        err_msg = "failed to init the socket " .. soc_err
+        return res, err_msg
+    end
+
+    sock:settimeout(config.timeout * 1000)
+
+    local ok, err = sock:connect(config.host, config.port)
+    if not ok then
+        err_msg = "failed to connect the TCP server: host[" .. config.host
+                  .. "] port[" .. tostring(config.port) .. "] err: " .. err
+        return res, err_msg
+    end
+
+    if config.tls then
+        ok, err = sock:sslhandshake(true, config.tls_options, false)
+        if not ok then
+            return false, "failed to to perform TLS handshake to TCP server: host["
+                          .. config.host .. "] port[" .. tostring(config.port) .. "] err: " .. err
+        end
+    end
+
+    table.insert(data, "\n")
+    local bytes, err = sock:send(data)
+    if not bytes then
+        sock:close()
+        err_msg = "failed to send data to TCP server: host[" .. config.host
+                  .. "] port[" .. tostring(config.port) .. "] err: " .. err
+        return res, err_msg
+    end
+
+    sock:setkeepalive(config.keepalive * 1000)
+    return true
+end
+
+
+local function process()
+    if not loaded_flag then
+        local code, body = plugin_metadata.get(plugin_name)
```

Comment:
this way is wrong, please take a look at: https://github.com/apache/apisix/blob/master/apisix/plugins/http-logger.lua#L166